src/analysis/normed_space/M_structure.lean

```diff
@@ -0,0 +1,439 @@
+      abel,
+    end,
+    apply e,
+  end,
+  sup_le := λ P Q R, begin
+    intros h₁ h₂,
+    have e₁: ↑P = ↑P * ↑R := h₁,
+    have e₂: ↑Q = ↑Q * ↑R := h₂,
+    have e:  ↑(P ⊔ Q) = ↑(P ⊔ Q) * ↑R := by rw [coe_sup, ← add_sub, add_mul, sub_mul, mul_assoc,
+       ← e₂, ← e₁],
+    apply e,
+  end,
+  inf_le_left := λ P Q, begin
+    have e: ↑(P ⊓ Q) = ↑(P ⊓ Q) * ↑P := by rw [coe_inf, mul_assoc,
+      commute.eq (Lproj_commute Q.prop P.prop), ← mul_assoc, ← sq, (projection_def P.prop.left)],
+    apply e,
+  end,
+  inf_le_right := λ P Q, begin
+    have e: ↑(P ⊓ Q) = ↑(P ⊓ Q) * ↑Q := by
+      rw [coe_inf, mul_assoc,  ← sq, (projection_def Q.prop.left)],
+    apply e,
+  end,
+  le_inf := λ P Q R, begin
+    intros h₁ h₂,
+    have e₁: ↑P = ↑P * ↑Q := h₁,
+    have e: ↑P =  ↑P * ↑(Q ⊓ R) := begin
+      rw [coe_inf, ← mul_assoc, ← e₁],
+      apply h₂,
+    end,
+    apply e,
+  end,
+  le_sup_inf := λ P Q R, begin
+    have e₁: ↑((P ⊔ Q) ⊓ (P ⊔ R)) = ↑P + ↑Q * ↑R * ↑Pᶜ := by rw [coe_inf, coe_sup, coe_sup,
+      ← add_sub, ← add_sub, compl_mul_left, compl_mul_left, add_mul,
+      mul_add, commute.eq (Lproj_commute Pᶜ.prop Q.prop), mul_add, ← mul_assoc, mul_assoc ↑Q,
+      commute.eq (Lproj_commute Pᶜ.prop P.prop), compl_orthog, zero_mul, mul_zero, zero_add,
+      add_zero, ← mul_assoc, mul_assoc ↑Q, ←sq, ← sq, projection_def P.prop.left,
+      projection_def Pᶜ.prop.left, mul_assoc, commute.eq (Lproj_commute Pᶜ.prop R.prop),
+      ← mul_assoc],
+    have e₂ : ↑((P ⊔ Q) ⊓ (P ⊔ R)) * ↑(P ⊔ Q ⊓ R) = ↑P + ↑Q * ↑R * ↑Pᶜ := by rw [coe_inf, coe_sup,
+      coe_sup, coe_sup, ← add_sub, ← add_sub, ← add_sub, compl_mul_left, compl_mul_left,
+      compl_mul_left, commute.eq (Lproj_commute Pᶜ.prop (Q⊓R).prop), coe_inf, mul_assoc,
+      distrib_lattice_lemma, commute.eq (Lproj_commute Q.prop R.prop), distrib_lattice_lemma],
+    have e: ↑((P ⊔ Q) ⊓ (P ⊔ R)) = ↑((P ⊔ Q) ⊓ (P ⊔ R)) * ↑(P ⊔ Q ⊓ R) := by rw [e₂, e₁],
+    apply e,
+  end,
+  .. is_Lprojection.subtype.has_inf,
+  .. is_Lprojection.subtype.has_sup,
+  .. is_Lprojection.subtype.partial_order }
+
+instance : boolean_algebra (subtype (is_Lprojection  : (X →L[𝕜] X) → Prop)) :=
+{ sup_inf_sdiff := λ P Q, begin
+    apply subtype.eq,
+    simp only [subtype.val_eq_coe, coe_sup, coe_inf, coe_sdiff],
+    rw [mul_assoc, ← mul_assoc ↑Q, commute.eq (Lproj_commute Q.prop P.prop), mul_assoc ↑P ↑Q,
+      ← coe_compl, compl_orthog, mul_zero, mul_zero, sub_zero, ← mul_add, coe_compl,
+      add_sub_cancel'_right, mul_one],
+  end,
+  inf_inf_sdiff := λ P Q, begin
+    apply subtype.eq,
+    simp only [subtype.val_eq_coe, coe_inf, coe_sdiff, coe_bot],
+    rw [mul_assoc, ← mul_assoc ↑Q, commute.eq (Lproj_commute Q.prop P.prop), ← coe_compl, mul_assoc,
+      compl_orthog, mul_zero, mul_zero],
+  end,
+  inf_compl_le_bot := λ P, begin
+    apply eq.le,
+    apply subtype.eq,
+    simp only [subtype.val_eq_coe, coe_inf, coe_compl, coe_bot],
+    rw [← coe_compl, compl_orthog],
+  end,
```

Comment:
I would usually pull out any "leading apply"s:
```suggestion
  inf_compl_le_bot := λ P, eq.le (subtype.eq begin
    simp only [subtype.val_eq_coe, coe_inf, coe_compl, coe_bot],
    rw [← coe_compl, compl_orthog],
  end),
```
Not strictly necessary, but it's usually a good habit to "golf" a bit further towards term mode.

(and elsewhere)

Reply:
Thanks. I think I've fixed all of the occurrences of this now?